### PR TITLE
Create test modules for reboot_before_migration  and stop_grub_timeout

### PR DESCRIPTION
**Pull request by hjluo**
- Description:
  * Create test modules for reboot_before_migration and stop_grub_timeout.
  
- Related ticket: https://progress.opensuse.org/issues/132950
- Needles: N/A
- Verification run: 
  * https://openqa.suse.de/tests/12615294
  * https://openqa.suse.de/tests/12615295
  * https://openqa.suse.de/tests/12615296
  * https://openqa.suse.de/tests/12615297

### CODE REVIEW

File: tests/migration/reboot_to_migration.pm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#       Just trigger reboot action, afterwards tests will be
+#       incepted by later test modules, such as tests in
+#       load_boot_tests or wait_boot in setup_zdup.pm
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
```
Comment (lemon-suse) [inline tests/migration/reboot_to_migration.pm:16]: We can remove it if not used.

File: tests/migration/reboot_to_migration.pm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# Summary: Reboot machine to perform upgrade
+#       Just trigger reboot action, afterwards tests will be
+#       incepted by later test modules, such as tests in
+#       load_boot_tests or wait_boot in setup_zdup.pm
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
```
Comment (lemon-suse) [inline tests/migration/reboot_to_migration.pm:15]: Only include functions used in this module.

File: tests/migration/reboot_to_migration.pm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    # Mark the hdd has been patched
+    set_var('PATCHED_SYSTEM', 1);
+
+    assert_script_run "sync", 300;
+    power_action('reboot', textmode => 1, keepconsole => 1);
+
+    assert_screen('inst-bootmenu', 300) unless (is_s390x || is_pvm);
+
+    send_key 'up' if is_x86_64;
+    save_screenshot;
```
Comment (lemon-suse) [inline tests/migration/reboot_to_migration.pm:36]: I think this is not needed, WDYT?

**Comment by lemon-suse** [comment]
Please provide VR.

File: tests/migration/reboot_to_migration.pm (hunk 2 above)
Comment (hjluo) [reply tests/migration/reboot_to_migration.pm:15]: OK.

File: tests/migration/reboot_to_migration.pm (hunk 3 above)
Comment (hjluo) [reply tests/migration/reboot_to_migration.pm]: removed.

File: tests/migration/reboot_to_migration.pm (hunk 1 above)
Comment (hjluo) [reply tests/migration/reboot_to_migration.pm]: yes. removed.

File: tests/migration/reboot_to_migration.pm [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Utils::Architectures;
+use Utils::Backends 'is_pvm';
+use power_action_utils 'power_action';
+use bootloader_setup 'stop_grub_timeout';
```
[the remark this answers is not in the document]
Comment (hjluo) [reply tests/migration/reboot_to_migration.pm]: ok, removed.

**Comment by hjluo** [comment]
@lemon-suse  Can you please take a look again? thanks a lot.

File: tests/migration/reboot_to_migration.pm (hunk 2 above)
Comment (lemon-suse) [reply tests/migration/reboot_to_migration.pm:15]: select_console,set_var, assert_screen include from testapi, you can double check others also.

**Review by lemon-suse — APPROVED** [review]

File: tests/migration/reboot_to_migration.pm (hunk 2 above)
Comment (hjluo) [reply tests/migration/reboot_to_migration.pm]: changed and verified https://openqa.suse.de/t12615294

**Pull request merged**